source/module_cell/print_cif.cpp

```diff
@@ -0,0 +1,169 @@
+#include "module_cell/unitcell.h"
+#include <cmath>   // for std::pow and std::round
+
+// Function to calculate the greatest common divisor
+long long gcd_func(long long a, long long b) {
+    return b == 0 ? a : gcd_func(b, a % b);
+}
+
+std::string floatToFraction(double number, double accuracy = 1e-9) {
+    // Determine the sign of the number
+    int sign = number < 0 ? -1 : 1;
+    number = std::abs(number);
+
+    // The maximum denominator is set to a large number to ensure precision
+    const long long max_denominator = 1000000;
+
+    // Start with a denominator of 1
+    long long denominator = 1;
+    long long numerator = std::round(number * denominator);
+
+    // Keep increasing the denominator to find a better approximation
+    double error;
+    do {
+        denominator *= 10;
+        numerator = std::round(number * denominator);
+        error = std::abs(number - (double)numerator / denominator);
+    } while (error > accuracy && denominator <= max_denominator);
+
+    // Simplify the fraction by dividing both numerator and denominator by their GCD
+    long long gcd = gcd_func(numerator, denominator);
+    numerator /= gcd;
+    denominator /= gcd;
+
+    // Apply the sign to the numerator
+    numerator *= sign;
+
+    if (denominator == 1) {
+        return std::to_string(numerator);
+    } else {
+        return std::to_string(numerator) + "/" + std::to_string(denominator);
+    }
+}
+
+void UnitCell::print_cell_cif(const std::string& fn) const
```

Comment:
I would sugges turing print_cell_cif as a free function independent of UnitCell, and make UnitCell as an input variable for the free function.